lib/account/account-manager.js

```diff
@@ -1,102 +1,171 @@
+ * @property {string} passwordSalt
+ * @property {string} mfaKey
  */
-class AccountManager {
+
+/**
+ * Contains the registration id needed for a user to
+ * continue on with the sign up process.
+ *
+ * @typeDef {Object} accountManager~completionInfo
+ * @property {string} accountId Unhased account id generated
+ */
+
+/**
+ * Contains the accountId an account holder needs to
+ * use the service.
+ *
+ * @typeDef {Object} accountManager~initiationInfo
+ * @property {string} accountId Unhased account id generated
+ */
+
+/**
+ * @param {Object} accoutService
+ * @param {Object} config
+ * @param {Object} hotp
+ * @param {Object} otDate
+ * @param {Object} random
+ * @param {Object} promise
+ * @return {Object} account manager object
+ */
+module.exports = function (accountService, config, hotp, otDate, promise, random) {
     /**
-     * @param {Object} accountService
-     * @param {Object} config
-     * @param {Object} hotp
-     * @param {Object} otDate
-     * @param {Object} password
-     * @param {Object} promise
-     * @param {Object} random
+     * Checks to see if the values for MFA are valid and will
+     * throw if they are not.
+     *
+     * @param {string} mfaKey
+     * @param {string} previousMfa
+     * @param {string} currentMfa
+     * @throws {Error}
      */
-    constructor(accountService, config, hotp, otDate, random, promise) {
-        this.accountService = accountService;
-        this.config = config;
-        this.hotp = hotp;
-        this.otDate = otDate;
-        this.random = random;
-        this.promise = promise;
+    function checkMfa(mfaKey, previousMfa, currentMfa) {
+        if (! hotp.verifyToken(mfaKey, previousMfa, 1)) {
+            throw new Error("Previous MFA Token did not validate");
+        }
+
+        if (! hotp.verifyToken(mfaKey, currentMfa)) {
+            throw new Error("Current MFA Token did not validate");
+        }
+    }
+
+
+    /**
+     * Finishes setting up the account and calls the Account Service
+     * to save an account file with the requested information.
+     *
+     * @param {string} accountId
+     * @param {Object} accountInfo
+     * @return {Promise.<accountManager~completionInfo>}
+     */
+    function setupAccountAsync(accountId, accountInfo) {
+        var options, regId;
+
+        accountInfo.accountId = accountId;
+        accountInfo.password = accountInfo.password;
+        regId = accountInfo.regId;
+        options = {
+            expires: otDate.now().plus(config.account.completeLifetime)
+        };
+
+        return accountService.completeAsync(accountInfo, options, regId);
     }
 
 
     /**
      * Validates the previous and current MFA values are valid and sends
-     * data off to Account service class for updating/putting data into
-     * the Account record.
+     * data off to the account service for updating/putting data into
+     * the account record.
      *
      * Password generation should follow this format:
```

Comment:
This bit of very useful information advice should be in the documentation (doc/) instead of buried in code.

What hashing mechanism is used?  If it is configurable, that should also be in the documentation.  As a hint, we may wish to supply the hashing method in the response for the user.

Consider:  Should we replace this hashing with pbkdf2 so it is more secure?